### DISCUSSION TITLE
Is it possible to execute command on pull in chezmoi external file?

### DISCUSSION THREAD

**Discussion by Demianeen** [twpayne/chezmoi#3664] [Q&A]
I recently started chezmoi and I absolutely love how I automated everything related to dotfiles)
I recently stumbled into another gem here, `.chezmoiexternal` file. But I found one limitation, I can't run command after pull, for example to build font and put it into right place alongside map for sketchybar. 

I stumbled on run_onchange and run_after. But then it basically builds the font on each `chezmoi apply` script, which takes ages.

Was anyone able to automate this?

**Comment by bradenhilton** [comment]
If it's relatively quick/cheap to detect when the important file(s) in an external have changed, you can still use a `run_onchange_*` script. One of the methods to achieve this is to generate a hash of the relevant file(s) and place the hash somewhere in the script. When the file(s) change, the hash will change, which will cause the script to run again.

See https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#run-a-script-when-the-contents-of-another-file-changes
